src/wp-admin/includes/image.php

```diff
@@ -914,9 +913,52 @@ function wp_generate_attachment_metadata( $attachment_id, $file ) {
 					// Save the meta data before any image post-processing errors could happen.
 					wp_update_attachment_metadata( $attachment_id, $metadata );
 
-					// Create sub-sizes saving the image meta after each.
-					$metadata = _wp_make_subsizes( $merged_sizes, $image_file, $metadata, $attachment_id, '' );
+					$primary_mime_type = isset( $metadata['sizes']['full']['mime-type'] ) ? $metadata['sizes']['full']['mime-type'] : 'image/jpeg';
```

Comment:
replace this block with 
`list( $primary_mime_type, $additional_mime_types ) = _wp_get_primary_and_additional_mime_types( $file, $attachment_id );` 
which we use elsewhere. that will give you the primary and secondary mime types (if any)

Reply:
done!